### ISSUE TITLE
Qlot Breaks on Error

### ISSUE THREAD

**Issue by daninus14** [fukamachi/qlot#275]
## Describe the bug

When doing an install (from a phone hotspot) qlot fails with the following message:

```
Unexpected error: The condition Socket error in "connect": EINTR (Interrupted system call) occurred with errno: 0.
This could be a bug in Qlot.
Report it at https://github.com/fukamachi/qlot/issues/new/choose.
Please attach the stack trace dumped to '/var/folders/nc/ln1v4ckx4yn201k8mrkwflw80000gn/T/qlot-error-JU0JWO19.log'.
```

I can run `qlot install` again many times, and the whole process works. But I would expect it to work. This may be because of some parallel download failing to connect? I would expect qlot to try again and not stop the other downloads which are working.

## Reproducible steps

`qlot init` and `qlot install` maybe from a not so great internet connection with a project with a bunch of dependencies.

## Error messages
```
Unexpected error: The condition Socket error in "connect": EINTR (Interrupted system call) occurred with errno: 0.
This could be a bug in Qlot.
Report it at https://github.com/fukamachi/qlot/issues/new/choose.
Please attach the stack trace dumped to '/var/folders/nc/ln1v4ckx4yn201k8mrkwflw80000gn/T/qlot-error-JU0JWO19.log'.
```

```
Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {10014A8143}>
0: ((LAMBDA NIL :IN UIOP/IMAGE:PRINT-BACKTRACE))
1: ((FLET "THUNK" :IN UIOP/STREAM:CALL-WITH-SAFE-IO-SYNTAX))
2: (SB-IMPL::%WITH-STANDARD-IO-SYNTAX #<FUNCTION (FLET "THUNK" :IN UIOP/STREAM:CALL-WITH-SAFE-IO-SYNTAX) {107DCCB}>)
3: (UIOP/STREAM:CALL-WITH-SAFE-IO-SYNTAX #<FUNCTION (LAMBDA NIL :IN UIOP/IMAGE:PRINT-BACKTRACE) {1004BC91BB}> :PACKAGE :CL)
4: (UIOP/IMAGE:PRINT-CONDITION-BACKTRACE #<USOCKET:UNKNOWN-ERROR {1004738033}> :STREAM #<SB-SYS:FD-STREAM for "file /var/folders/nc/ln1v4ckx4yn201k8mrkwflw80000gn/T/qlot-error-JU0JWO19.log" {1004BC9053}> :COUNT NIL)
5: ((FLET "BEFORE29" :IN QLOT/CLI:QLOT-COMMAND) #<SB-SYS:FD-STREAM for "file /var/folders/nc/ln1v4ckx4yn201k8mrkwflw80000gn/T/qlot-error-JU0JWO19.log" {1004BC9053}> #P"/var/folders/nc/ln1v4ckx4yn201k8mrkwflw80000gn/T/qlot-error-JU0JWO19.log")
6: (UIOP/STREAM:CALL-WITH-TEMPORARY-FILE #<FUNCTION (FLET "BEFORE29" :IN QLOT/CLI:QLOT-COMMAND) {107E14B}> :WANT-STREAM-P T :WANT-PATHNAME-P T :DIRECTION :OUTPUT :KEEP T :AFTER NIL :DIRECTORY NIL :TYPE "log" :PREFIX "qlot-error-" :SUFFIX "" :ELEMENT-TYPE NIL :EXTERNAL-FORMAT NIL)
7: ((LAMBDA (QLOT/CLI::C) :IN QLOT/CLI:QLOT-COMMAND) #<USOCKET:UNKNOWN-ERROR {1004738033}>)
8: (SB-KERNEL::%SIGNAL #<USOCKET:UNKNOWN-ERROR {1004738033}>)
9: (ERROR #<USOCKET:UNKNOWN-ERROR {1004738033}>)
10: (LPARALLEL.KERNEL-UTIL::RECEIVE-RESULTS #S(LPARALLEL.KERNEL:CHANNEL :QUEUE #S(LPARALLEL.CONS-QUEUE:CONS-QUEUE :IMPL #S(LPARALLEL.RAW-QUEUE:RAW-QUEUE :HEAD NIL :TAIL NIL) :LOCK #<SB-THREAD:MUTEX "Anonymous lock" (free)> :CVAR #<SB-THREAD:WAITQUEUE Anonymous condition variable {10032DF0A3}>) :KERNEL #<LPARALLEL.KERNEL:KERNEL :NAME "lparallel" :WORKER-COUNT 4 :USE-CALLER NIL :ALIVE T :SPIN-COUNT 2000 {10032DE013}>) #<unavailable argument> #<unavailable argument>)
11: (LPARALLEL.COGNATE::PMAP-ITERATE #<FUNCTION MAPC> #<FUNCTION (LAMBDA (QLOT/PROGRESS::JOB) :IN QLOT/PROGRESS:RUN-IN-PARALLEL) {10032DEE4B}> ((#<QLOT/SOURCE/GITHUB:SOURCE-GITHUB fukamachi/caveman a644adbc188cb6b929825098cea9092813156bfe {100177FB03}> #<QLOT/SOURCE/QL:SOURCE-QL #1=https://beta.quicklisp.org/dist/quicklisp.txt ql-2023-10-21 {1001633153}> #<QLOT/SOURCE/QL:SOURCE-QL #1# ql-2023-10-21 {100171D603}> #<QLOT/SOURCE/QL:SOURCE-QL #1# ql-2023-10-21 {100171D613}> #<QLOT/SOURCE/QL:SOURCE-QL #1# ql-2023-10-21 {10017C0223}> #<QLOT/SOURCE/QL:SOURCE-QL #1# ql-2023-10-21 {1001633163}> #<QLOT/SOURCE/GITHUB:SOURCE-GITHUB nobody-famous/alive-lsp <undetermined> {100171D633}> #<QLOT/SOURCE/QL:SOURCE-QL #1# ql-2023-10-21 {10017C0333}> #<QLOT/SOURCE/GITHUB:SOURCE-GITHUB fukamachi/dexador d2717a409dd46609531856c5001b1d4f83de5536 {10016331B3}> #<QLOT/SOURCE/QL:SOURCE-QL #1# ql-2023-10-21 {100171D673}> #<QLOT/SOURCE/GITHUB:SOURCE-GITHUB fukamachi/prove 5d71f02795b89e36f34e8c7d50e69b67ec6ca2de {10017C0573}> #<QLOT/SOURCE/GITHUB:SOURCE-GITHUB fukamachi/woo 7f5219c55d49190f5ae17b123a8729b31c5d706e {10016331C3}> ...)) 13 4)
12: (LPARALLEL.COGNATE:PMAPC #<FUNCTION (LAMBDA (QLOT/PROGRESS::JOB) :IN QLOT/PROGRESS:RUN-IN-PARALLEL) {10032DEE4B}> (#<QLOT/SOURCE/GITHUB:SOURCE-GITHUB fukamachi/caveman a644adbc188cb6b929825098cea9092813156bfe {100177FB03}> #<QLOT/SOURCE/QL:SOURCE-QL #1=https://beta.quicklisp.org/dist/quicklisp.txt ql-2023-10-21 {1001633153}> #<QLOT/SOURCE/QL:SOURCE-QL #1# ql-2023-10-21 {100171D603}> #<QLOT/SOURCE/QL:SOURCE-QL #1# ql-2023-10-21 {100171D613}> #<QLOT/SOURCE/QL:SOURCE-QL #1# ql-2023-10-21 {10017C0223}> #<QLOT/SOURCE/QL:SOURCE-QL #1# ql-2023-10-21 {1001633163}> #<QLOT/SOURCE/GITHUB:SOURCE-GITHUB nobody-famous/alive-lsp <undetermined> {100171D633}> #<QLOT/SOURCE/QL:SOURCE-QL #1# ql-2023-10-21 {10017C0333}> #<QLOT/SOURCE/GITHUB:SOURCE-GITHUB fukamachi/dexador d2717a409dd46609531856c5001b1d4f83de5536 {10016331B3}> #<QLOT/SOURCE/QL:SOURCE-QL #1# ql-2023-10-21 {100171D673}> #<QLOT/SOURCE/GITHUB:SOURCE-GITHUB fukamachi/prove 5d71f02795b89e36f34e8c7d50e69b67ec6ca2de {10017C0573}> #<QLOT/SOURCE/GITHUB:SOURCE-GITHUB fukamachi/woo 7f5219c55d49190f5ae17b123a8729b31c5d706e {10016331C3}> ...))
13: (QLOT/PROGRESS:RUN-IN-PARALLEL #<FUNCTION (LAMBDA (QLOT/INSTALL::SOURCE) :IN QLOT/INSTALL::APPLY-QLFILE-TO-QLHOME) {100320419B}> (#<QLOT/SOURCE/GITHUB:SOURCE-GITHUB fukamachi/caveman a644adbc188cb6b929825098cea9092813156bfe {100177FB03}> #<QLOT/SOURCE/QL:SOURCE-QL #1=https://beta.quicklisp.org/dist/quicklisp.txt ql-2023-10-21 {1001633153}> #<QLOT/SOURCE/QL:SOURCE-QL #1# ql-2023-10-21 {100171D603}> #<QLOT/SOURCE/QL:SOURCE-QL #1# ql-2023-10-21 {100171D613}> #<QLOT/SOURCE/QL:SOURCE-QL #1# ql-2023-10-21 {10017C0223}> #<QLOT/SOURCE/QL:SOURCE-QL #1# ql-2023-10-21 {1001633163}> #<QLOT/SOURCE/GITHUB:SOURCE-GITHUB nobody-famous/alive-lsp <undetermined> {100171D633}> #<QLOT/SOURCE/QL:SOURCE-QL #1# ql-2023-10-21 {10017C0333}> #<QLOT/SOURCE/GITHUB:SOURCE-GITHUB fukamachi/dexador d2717a409dd46609531856c5001b1d4f83de5536 {10016331B3}> #<QLOT/SOURCE/QL:SOURCE-QL #1# ql-2023-10-21 {100171D673}> #<QLOT/SOURCE/GITHUB:SOURCE-GITHUB fukamachi/prove 5d71f02795b89e36f34e8c7d50e69b67ec6ca2de {10017C0573}> #<QLOT/SOURCE/GITHUB:SOURCE-GITHUB fukamachi/woo 7f5219c55d49190f5ae17b123a8729b31c5d706e {10016331C3}> ...) :CONCURRENCY 4 :JOB-HEADER-FN #<FUNCTION (LAMBDA (QLOT/INSTALL::SOURCE) :IN QLOT/INSTALL::APPLY-QLFILE-TO-QLHOME) {10032041CB}> :FAILED-FN #<FUNCTION (LAMBDA NIL :IN QLOT/INSTALL::APPLY-QLFILE-TO-QLHOME) {5596281B}>)
.
.
.
Above backtrace due to this condition:
The condition Socket error in "connect": EINTR (Interrupted system call) occurred with errno: 0.

```
## Versions:
 - Qlot version: 1.5.10
 - Where did you get it from:  Automatic Installer
 - OS: macos
 - Lisp:  SBCL 2.4.1

**Comment by fukamachi** [comment]
It sounds reasonable. I'll think about adding some retry functionality.

**Comment by fukamachi** [comment]
Fixed in [1.5.11](https://github.com/fukamachi/qlot/releases/tag/1.5.11).
